apps/web/middleware.ts

```diff
@@ -79,6 +108,8 @@ const middleware: NextMiddleware = async (req) => {
 
 export const config = {
   matcher: [
+    "/",
+    "/:path*",
```

Comment:
That would significantly increase the invocations of serverless functions(as it's matching all the requests coming in). Can we achieve the same effect using next.config.js rewrites ?

Reply:
Managed to remove matcher `"/"` as the one with `"/:path*"` covers it, but unfortunately, we can't do anything for that one left, because we need to be able to distinguish a user from a team as `/username` and `/teamslug` are both accessible equally. I would have preferred to leave it as we had `/username` and `/team/teamslug` but the decision was to be able to support that at all costs if I remember correctly.

Reply:
I know this is necessary for orgs to function but expanding on @hariombalhara comment. This WILL increase our server less function invocations by a lot. This will be a big hit to self-hosted users billing in Vercel. We're fortunate enough to be sponsored but can't say the same for Self-hosted users. cc @keithwillcode @PeerRich and @baileypumfleet for visibility.

Reply:
Being discussed separately. So resolving this.